Remotion/Web/Core/UI/Controls/RendererBase.cs

```diff
@@ -144,13 +143,15 @@ protected IResourceManager GetResourceManager (Type localResourcesType, IResourc
       ArgumentUtility.CheckNotNull ("localResourcesType", localResourcesType);
       ArgumentUtility.CheckNotNull ("controlResourceManager", controlResourceManager);
 
-      var cachedResourceManager = _resourceManagerCache.GetOrAdd (
-          Tuple.Create (localResourcesType, controlResourceManager),
-          key => new Lazy<IResourceManager> (
-              () => ResourceManagerSet.Create (key.Item2, _globalizationService.GetResourceManager (key.Item1)),
-              LazyThreadSafetyMode.ExecutionAndPublication));
+      var result = _resourceManagerCache
+          .GetOrAdd (
+              localResourcesType,
+              _ => new ConditionalWeakTable<IResourceManager, Lazy<IResourceManager>>())
+          .GetValue (
+              controlResourceManager,
+              mgr => new Lazy<IResourceManager> (() => ResourceManagerSet.Create (mgr, _globalizationService.GetResourceManager (localResourcesType))));
```

Comment:
I was skeptical at first if we need the `Lazy<T>` here, because we always directly create the value in line 154. But it turns out doing things like this has the benefit of thread safety as two contending threads will only create two `Lazy<T>` objects one of which is thrown away by the dictionary and then when accessing the value on of the threads will block.